src/app/lists/dataview.tsx

```diff
@@ -0,0 +1,199 @@
+    const [modal, setModal] = useState<Datum | null | undefined>(undefined);
+    const [toDelete, setToDelete] = useState<number | null>(null);
+    // need another setState otherwise the modal content changes before the close animation finishes
+    const [toDeleteModalOpen, setToDeleteModalOpen] = useState<boolean>(false);
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    if (rawData === null) {
+        throw new Error("No data found");
+    }
+
+    const remapTooltips = (
+        row: ListRow
+    ): Datum & { unmappedTooltips: { [key: string]: string } } => {
+        const data: { [key: string]: string } = {
+            item_name: row.item_name,
+            primary_key: row.primary_key,
+        };
+        const tooltips: { [key: string]: string } = {};
+        const unmappedTooltips: { [key: string]: string } = {};
+
+        for (const [key, value] of Object.entries(row)) {
+            if (key.endsWith("quantity") && value) {
+                data[key] = value;
+            } else if (key.endsWith("notes") && value) {
+                tooltips[key.replace("notes", "quantity")] = value;
+                unmappedTooltips[key] = value;
+            }
+        }
+
+        return {
+            data,
+            tooltips,
+            unmappedTooltips,
+        };
+    };
+
+    const dataAndTooltips = rawData?.map(remapTooltips);
+
+    // extract header keys
+    const toggleableHeaders = headers.map(([key]) => key);
+    // removing description header from Toggleable Headers using shift
+    toggleableHeaders.shift();
+
+    const onEdit = (index: number): void => {
+        const row = rawData[index];
+
+        const editData = remapTooltips(row!);
+
+        setModal({
+            data: editData.data,
+            tooltips: editData.unmappedTooltips,
+        });
+    };
+
+    const onDelete = async (index: number): Promise<void> => {
+        setToDelete(index);
+        setToDeleteModalOpen(true);
+    };
+
+    return (
+        <>
+            <ConfirmDialog
+                message={`Are you sure you want to delete ${
+                    toDelete ? rawData[toDelete].item_name : ""
+                }?`}
+                isOpen={toDeleteModalOpen}
+                onConfirm={async () => {
+                    if (toDelete !== null) {
+                        const data = rawData[toDelete];
+                        const response = await supabase
+                            .from("lists")
+                            .delete()
+                            .match({ primary_key: data.primary_key });
+
+                        if (Math.floor(response.status / 100) !== 2) {
+                            setErrorMsg(
+                                `${response.status}: ${response.statusText} -- data: ${response.data}`
+                            );
+                        } else {
+                            setToDeleteModalOpen(false);
+                            window.location.reload();
```

Comment:
why are we reloading here?

Reply:
So we refetch data so the updated data is reflected in the UI. For some reason, `useRouter().refresh()` doesn't seem to work...

Reply:
hmmm this may be a sign that a new page can do the editting/adding directly, with the modal only viewing, so navigating back to the data page 'automatically' updates...
but this is a fine solution for the time being!